### PR TITLE
931 fix selflock bug

### LINKED ISSUE #931: [Users Dashboard] Admin can lock himself from the app
**Environment:** Windows 10
**Reproducible:** always
**Build found:** develop

**Preconditions**
1. Site https://zero-waste-staging.onrender.com/en is opened.
2. Admin is logged in
3. 'Users' page is opened on Admin dashboard

**Steps to reproduce**
1. Click on 'lock icon' on the user you currently logged in
2. Confirm action

**Actual result**
User locked himself from the app

**Expected result**
User cannot block himself

https://github.com/user-attachments/assets/eeb2743e-2d11-4020-be3e-9a4634ea6418

### PR DESCRIPTION
**Pull request by DanielVajnagi**
dev
## Resolves

#931 

## Code reviewers

- [x] @alexy78 

### Second Level Review

- [ ] @loqimean 
## Summary of issue

Admin can ban himself 

## Summary of change

Removed this function

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

**Review by alexy78 — APPROVED** [review]

File: app/controllers/account/users_controller.rb [hunk 1]
```diff
@@ -43,6 +43,12 @@ def create
   def update
     @user = resource
 
+    if blocked_admin?
+      flash[:alert] = t("errors.messages.blocked_user_cannot_be_admin")
+      redirect_to account_user_path(id: @user)
+      return
+    end
```
Comment (loqimean) [inline app/controllers/account/users_controller.rb:50]: 1. біфор екшин, ну я ж казав
2. навіщо перевіряти, чи користувач заблокований, чи ні? просто якщо юзер з парамсів - адмін, то редірект і заборона

**Review by loqimean — CHANGES_REQUESTED** [review]

File: app/controllers/account/users_controller.rb (hunk 1 above)
Comment (DanielVajnagi) [reply app/controllers/account/users_controller.rb]: пофіксав

File: app/controllers/account/users_controller.rb [hunk 2]
```diff
@@ -71,6 +72,14 @@ def user_params
     prms
   end
 
+  def blocking_admin
+    @user = resource
+    if params[:user][:blocked] && @user.role == "admin"
+      flash[:alert] = t("errors.messages.blocked_user_cannot_be_admin")
+      redirect_to account_user_path(id: @user)
```
Comment (loqimean) [inline app/controllers/account/users_controller.rb:79]:
```suggestion
    @user = resource

    if params.dig(:user, :blocked) && @user.admin?
      flash[:alert] = t("errors.messages.blocked_user_cannot_be_admin")

      redirect_to account_user_path(@user)
```

Comment (loqimean) [reply app/controllers/account/users_controller.rb:79]: guard style return

**Review by loqimean — APPROVED** [review]

**Pull request merged**